### PR TITLE
grt: fix crash

### PR DESCRIPTION
**Pull request by eder-matheus**
Fixes https://github.com/The-OpenROAD-Project/OpenROAD/issues/774

### CODE REVIEW

File: src/grt/src/GlobalRouter.cpp [hunk 1]
```diff
@@ -2898,8 +2898,10 @@ int GlobalRouter::findInstancesObstructions(
     odb::Rect& dieArea,
     const std::vector<int>& layerExtensions)
 {
+  const int max_out_of_die_warn = 25;
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:2901]: Messages are already capped at 1000.  Is there a need for this lower limit?

File: src/grt/src/fastroute/src/maze.cpp [hunk 2]
```diff
@@ -1044,7 +1044,6 @@ void reInitTree(int netID)
     if (treeedge->len > 0) {
       delete[] treeedge->route.gridsX;
       delete[] treeedge->route.gridsY;
-      delete[] treeedge->route.gridsL;
```
Comment (maliberty) [inline src/grt/src/fastroute/src/maze.cpp:1047]: Why is this wrong?

Comment (eder-matheus) [reply src/grt/src/fastroute/src/maze.cpp:1047]: gridsL is not initialized until 3D routing, so I think it was deleting a nullptr.

Comment (maliberty) [reply src/grt/src/fastroute/src/maze.cpp:1047]: Ok though deleting a nullptr is not an error.  More likely it was uninitialized.

**Review by maliberty — APPROVED** [review]

**Pull request merged**